src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -220,10 +221,13 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
         'node-output-data',
         (data) => {
             if (data) {
-                setOutputDataMap((prev) => new Map([...prev, [data.nodeId, data.data]]));
+                const existingData = outputDataMap.get(data.nodeId);
+                if (!existingData || !isJsonEqual(existingData, data.data)) {
+                    setOutputDataMap((prev) => new Map([...prev, [data.nodeId, data.data]]));
+                }
             }
         },
-        [setOutputDataMap]
+        [unAnimate, outputDataMap, setOutputDataMap]
```

Comment:
These dependencies seem weird.

Reply:
Yeah this is kinda a problem -- without putting something here that forces it to update (which is what I used unAnimate for here), it doesn't properly update. I have no idea why, but using outputDataMap doesnt just work.